modules/create_persons.rb

```diff
@@ -0,0 +1,39 @@
+module CreatePerson
+  def create_person
+    print 'Do you want to create a student (1) Student, or a teacher (2)?  [input the number]: '
+    person_input = gets.chomp.to_i
+    case person_input
+    when 1 then create_student
+    when 2 then create_teacher
+    else
+      raise 'Please choose a valid option, number 1 or 2'
+    end
+  end
+
+  def create_student
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission_input = gets.chomp
+    parent_permission = parent_permission_input.upcase == 'Y'
+    student = Student.new(name, age, parent_permission: parent_permission)
+    @people << student
+    puts 'Student created successfully'
+    puts "\n"
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(name, age, specialization)
+    @people << teacher
+    puts 'Person created successfully'
+    puts "\n"
+  end
+end
```

Comment:
- [ ] All input and input validation requests should've been placed in the main.rb file and/or classes/methods which are responsible for requesting or validating input. please have another look at [this lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) which explains the best practices and the solid principle.

Reply:
Can you please elaborate? I am not getting what should i do?